endian: add loong64

bradfitz opened the pull request:
Commented in upstream PR too:
https://github.com/WireGuard/wireguard-go/pull/64/files#r1045140162

Updates tailscale/tailscale#6233

crawshaw commented on wgcfg/ip_test.go:
```diff
@@ -0,0 +1,113 @@
+package wgcfg_test
```
Please add SPDX-License-Identifier.

(I realize it's missing from ip.go, I'll add it there later in a rebase.)

kroppt replied:
Done.

kroppt commented:
I squashed it into one commit and signed it off.

crawshaw commented:
Thanks! When I PR wgcfg upstream (soon I hope), I'll keep this commit intact.

bradfitz committed:
endian: add loong64

Commented in upstream PR too:
https://github.com/WireGuard/wireguard-go/pull/64/files#r1045140162

Updates tailscale/tailscale#6233

Signed-off-by: Brad Fitzpatrick <bradfitz@tailscale.com>
